### ISSUE TITLE
Container build fails due to compile error

### ISSUE THREAD

**Issue by hhenkel** [plan44/vdcd#13]
Hi,

I cloned the repo and tried to build the container with the latest code:
```
mv -f src/p44utils/.deps/vdcd-iopin.Tpo src/p44utils/.deps/vdcd-iopin.Po
g++ -DHAVE_CONFIG_H -I.  -I ./src -I ./src/p44utils -I ./src/p44lrgraphics -I ./src/p44utils/thirdparty/civetweb -I ./src/p44utils/thirdparty/slre -idirafter ./src/p44utils/thirdparty -I ./src/p44vdc/pbuf/gen -I ./src/p44vdc/vdc_common -I ./src/p44vdc/behaviours -I ./src/p44vdc/deviceclasses/simpleio -I ./src/p44vdc/deviceclasses/custom -I ./src/p44vdc/deviceclasses/evaluator -I ./src/p44vdc/deviceclasses/enocean -I ./src/p44vdc/deviceclasses/eldat -I ./src/p44vdc/deviceclasses/zf -I ./src/p44vdc/deviceclasses/dali -I ./src/p44vdc/deviceclasses/hue -I ./src/p44vdc/deviceclasses/ledchain -D ENABLE_RRDB=0 -D ENABLE_P44FEATURES=0 -D ENABLE_RPIWS281X=0 -D ENABLE_OLA=0 -D ENABLE_PNG=0 -D ENABLE_UBUS=0  -D ENABLE_UWSC=0  -pthread -I/usr/include/json-c    -D_REENTRANT -D NO_SSL_DL=1  -D DISABLE_I2C=1   -g -O2 -MT src/p44utils/vdcd-jsoncomm.o -MD -MP -MF src/p44utils/.deps/vdcd-jsoncomm.Tpo -c -o src/p44utils/vdcd-jsoncomm.o `test -f 'src/p44utils/jsoncomm.cpp' || echo './'`src/p44utils/jsoncomm.cpp
mv -f src/p44utils/.deps/vdcd-jsoncomm.Tpo src/p44utils/.deps/vdcd-jsoncomm.Po
g++ -DHAVE_CONFIG_H -I.  -I ./src -I ./src/p44utils -I ./src/p44lrgraphics -I ./src/p44utils/thirdparty/civetweb -I ./src/p44utils/thirdparty/slre -idirafter ./src/p44utils/thirdparty -I ./src/p44vdc/pbuf/gen -I ./src/p44vdc/vdc_common -I ./src/p44vdc/behaviours -I ./src/p44vdc/deviceclasses/simpleio -I ./src/p44vdc/deviceclasses/custom -I ./src/p44vdc/deviceclasses/evaluator -I ./src/p44vdc/deviceclasses/enocean -I ./src/p44vdc/deviceclasses/eldat -I ./src/p44vdc/deviceclasses/zf -I ./src/p44vdc/deviceclasses/dali -I ./src/p44vdc/deviceclasses/hue -I ./src/p44vdc/deviceclasses/ledchain -D ENABLE_RRDB=0 -D ENABLE_P44FEATURES=0 -D ENABLE_RPIWS281X=0 -D ENABLE_OLA=0 -D ENABLE_PNG=0 -D ENABLE_UBUS=0  -D ENABLE_UWSC=0  -pthread -I/usr/include/json-c    -D_REENTRANT -D NO_SSL_DL=1  -D DISABLE_I2C=1   -g -O2 -MT src/p44utils/vdcd-jsonobject.o -MD -MP -MF src/p44utils/.deps/vdcd-jsonobject.Tpo -c -o src/p44utils/vdcd-jsonobject.o `test -f 'src/p44utils/jsonobject.cpp' || echo './'`src/p44utils/jsonobject.cpp
src/p44utils/jsonobject.cpp: In member function 'void p44::JsonObject::arrayDel(int, int)':
src/p44utils/jsonobject.cpp:444:5: error: 'json_object_array_del_idx' was not declared in this scope
     json_object_array_del_idx(json_obj, aAtIndex, aNumElements);
     ^~~~~~~~~~~~~~~~~~~~~~~~~
src/p44utils/jsonobject.cpp:444:5: note: suggested alternative: 'json_object_array_get_idx'
     json_object_array_del_idx(json_obj, aAtIndex, aNumElements);
     ^~~~~~~~~~~~~~~~~~~~~~~~~
     json_object_array_get_idx
Makefile:2789: recipe for target 'src/p44utils/vdcd-jsonobject.o' failed
make[1]: Leaving directory '/vdcd'
make[1]: *** [src/p44utils/vdcd-jsonobject.o] Error 1
Makefile:1226: recipe for target 'all' failed
make: *** [all] Error 2
error building at STEP "RUN cd ${DSROOR}/vdcd &&     autoreconf -i &&     ./configure &&     make clean && make all": error while running runtime: exit status 2
[2/2] STEP 1/4: FROM ubuntu:bionic
ERRO[0095] exit status 2
```

**Comment by plan44** [comment]
Hi @hhenkel, thanks for reporting this.

It seems `json_object_array_del_idx` was introduced in json-c only in version 0.13 (in 2017), but despite 4 years passed since, it apparently isn't yet widespread. The configure.ac for vdcd only request 0.10 at this time, that's why it passes autoconf checks and fails only later when compiling.

If you can install json-c 0.13 or later, then the build should work.

Or you can just comment out line 444 in jsonobject.cpp as a Q&D workaround, then json-c 0.10 will do. So far the only thing where this is used is when you want to delete array elements from a JSON variable in p44script.

For a real fix I have to check if I can increase the requirement to 0.13 for json-c in general, but if 0.13 is not easily available this might break to many build environments and I'll need to find a workaround.
Any feedback or ideas welcome!

**Comment by plan44** [comment]
Hi @hhenkel, I implemented a workaround in p44utils such that build should succeed when only json-c <0.13 is available, but still uses the newer version when 0.13 or later with `json_object_array_del_idx` is available.
(Note that for now, this is on the `master` branch only, so you need to set that in your Dockerfile)
